Add HackerRank

moee opened the pull request.

karlhorky commented:
This is free, correct?

karlhorky commented on README.md:
```diff
@@ -131,3 +131,7 @@ This list aims to be a curated set of high quality educational resources. The co
+
+#### [HackerRank](https://www.hackerrank.com/)
+Programming challenges with leaderboard. Used by various tech companies to hire engineers.  
+*(various languages and topics, C, C++, Haskell, Bash, ...)*
```
Here I would list out the topics as they present them on the website instead of individual languages, which could change.

So something like:
*(algorithms, artificial intelligence, functional programming, Bash)*

moee commented:
@karlhorky yes, afaik it is free for developers. if a company want to use hackerrank to test job candidates they need to subscribe.

karlhorky commented:
Great, then can you add the free label to this too? With that and the topics change, this should be good to merge.

moee commented:
Yeah, I changed the tags. I added *mixed* because they have some sort of subscription (for companies) and they do not state explicitly that it's always free for developers.

karlhorky commented:
Added with the `free` label and a slightly changed description. Thanks!
